### DISCUSSION TITLE
What's the kademlia peer expire strategy in kbucket?

### DISCUSSION THREAD

**Discussion by lynic** [libp2p/rust-libp2p#3042] [Q&A]
I'm trying out kademlia, and start/stop some random peers. Trying to understand how does kademlia routing table work, I have couple of questions:
* Once a peer's status is disconnected, how long will it be removed from kbuckets/routing table?
* Will a disconnected peer be advise to another peer's routing table?
* I found that even though the peer is destroyed, get_provider will still return this peer id.

**Comment by mxinden** [comment]
>     * Once a peer's status is disconnected, how long will it be removed from kbuckets/routing table?

It will be removed once we have a better peer available to replace the former.

>     * Will a disconnected peer be advise to another peer's routing table?

Yes. If I recall correctly.

>     * I found that even though the peer is destroyed, get_provider will still return this peer id.

See above.

Hope this is of some help.
